kornia/augmentation/random_generator/_2d/planckian_jitter.py

```diff
@@ -0,0 +1,45 @@
+from functools import partial
+from typing import Dict, List, Optional
+
+import torch
+from torch.distributions import Uniform
+
+from kornia.augmentation.random_generator.base import RandomGeneratorBase
+from kornia.augmentation.utils import (
+    _adapted_rsampling,
+    _adapted_uniform,
+    _common_param_check,
+    _joint_range_check,
+    _range_bound,
+)
+from kornia.utils.helpers import _deprecated, _extract_device_dtype
+
+
+class PlanckianJitterGenerator(RandomGeneratorBase):
+
+    r"""Generate random color jitter parameters for a batch of images
+    """
+
+    def __init__(self, domain: List[int]) -> None:
+        super().__init__()
+        self.domain = domain
+
+    def make_samplers(self, device: torch.device, dtype: torch.dtype) -> None:
+
+        idx_range: torch.Tensor = _range_bound(self.domain,
+                                               'idx_range',
+                                               device=device, dtype=dtype)
+
+        _joint_range_check(idx_range, 'idx_range', (0, self.domain[1]))
+        self.pl_idx_dist = Uniform(idx_range[0], idx_range[1], validate_args=False)
```

Comment:
would this work if my input is `[1, 3, 5]` or `[3, 2, 1]`? I think we shall generate a mask `[0, 0, 1]` to perform the selection.

Reply:
Yes, this is taken care of in the code below
https://github.com/kornia/kornia/blob/c36efdcac422e41a568bc7e8104627206e96ca87/kornia/augmentation/_2d/intensity/planckian_jitter.py#L113

Let me know if you this is not a good way to do.

* For the selected indices say `[1, 3, 5]`, I filter these values in the code below
https://github.com/kornia/kornia/blob/c36efdcac422e41a568bc7e8104627206e96ca87/kornia/augmentation/_2d/intensity/planckian_jitter.py#L113

and then we generate the random indices between `0 - self.pl.shape[0]` only, so it should work
https://github.com/kornia/kornia/blob/c36efdcac422e41a568bc7e8104627206e96ca87/kornia/augmentation/_2d/intensity/planckian_jitter.py#L117

Reply:
Will need some tests on this then.